### PR TITLE
fix: client side validation and copy

### PR DESCRIPTION
**Pull request by tnunamak**

### CODE REVIEW

File: components/prompt/Generator.js [hunk 1]
```diff
@@ -18,11 +18,7 @@ const Generator = ({ authToken, email, onSubmit }) => {
 
   const handleSubmit = async (event) => {
     event.preventDefault();
-    setIsLoading(true);
-    setValidPrompt(meRegex.test(prompt));
-
     if (!validPrompt) {
-      setIsLoading(false);
```
Comment (tnunamak) [inline components/prompt/Generator.js:25]: Does this seem okay to you?

File: components/prompt/Generator.js [hunk 2]
```diff
@@ -18,11 +18,7 @@ const Generator = ({ authToken, email, onSubmit }) => {
 
   const handleSubmit = async (event) => {
     event.preventDefault();
-    setIsLoading(true);
```
Comment (dmitryvlass) [inline components/prompt/Generator.js:21]: Why did you remove the loading state for the submitting? Did you want to move it just below the api call?

File: components/prompt/Generator.js [hunk 3]
```diff
@@ -92,13 +86,13 @@ const Generator = ({ authToken, email, onSubmit }) => {
...
           ) : (
-            <>Create {GENERATED_SAMPLES} images (~2 mins)</>
+            <>Create {GENERATED_SAMPLES} images (~7 mins)</>
           )}
         </button>
       </form>
 
       {/* regex error */}
-      {!validPrompt && (
+      {!validPrompt && prompt.length > 20 && (
```
Comment (dmitryvlass) [inline components/prompt/Generator.js:95]: Do we really need to have prompt length validation? What if user tries to request "Obama eats apple" (16 symbols)?

Comment (dmitryvlass) [reply components/prompt/Generator.js:95]: I suggest to add one more state variable "isSubmitted", and validate the prompt if user tried to submit the form

File: components/prompt/Generator.js (hunk 2 above)
Comment (tnunamak) [reply components/prompt/Generator.js:21]: Yes, thanks for catching that

File: components/prompt/Generator.js (hunk 3 above)
Comment (callumflack) [inline components/prompt/Generator.js:95]: @tnunamak nice, this is better than having it in the useEffect…

File: components/prompt/Generator.js (hunk 1 above)
Comment (callumflack) [reply components/prompt/Generator.js:25]: I think we still want to validate it during submission as a user could use less than 20 characters, which means this would fail w/o apparent reason.

FYI, I also use a timeout during `finally` to allow the UI to show "thought time" to make the flow understandable and trustworthy to users (not that you've changed that but just so you know my intention)

File: components/prompt/Generator.js (hunk 3 above)
Comment (tnunamak) [reply components/prompt/Generator.js:95]: @callumflack @dmitryvlass we can do this, but currently, they will not see the red error message for client-side validation failures because the button is disabled for invalid prompts

Comment (callumflack) [reply components/prompt/Generator.js:97]: I originally did this to only show the prompt error _after_ some typing, not straight away, as this will be annoying for a user.

If they use less than 20 characters without using "me", then `handleSubmit` handles it.

This way we cover prompt validation lazily (on input) and on submit (edge cases eg less than 20)

**Review by callumflack — APPROVED** [review]

File: components/prompt/Generator.js (hunk 3 above)
Comment (dmitryvlass) [reply components/prompt/Generator.js:97]: So we have an unhandled error, so user doesn't know why they can't submit the prompt. But I am fine with that as soon as it is a demo app

image

**Review by dmitryvlass — APPROVED** [review]

File: components/prompt/Generator.js (hunk 3 above)
Comment (callumflack) [reply components/prompt/Generator.js:97]: Oh. Then I agree with you. We want that feedback to let users know what's going on.
